### DISCUSSION TITLE
.zshrc:source:76: permission denied:

### DISCUSSION THREAD

**Discussion by kevinlockett** [ohmyzsh/ohmyzsh#10257] [Help] [answered]
I am a git newbie and think I broke my terminal.  I ran [sudo -s] and saw the following code (I has substituted USER for my name below:
******************************************************************************
[oh-my-zsh] Insecure completion-dependent directories detected:
dr--r--r--  12 USER USER  4096 Sep 27 14:39 /home/USER/.oh-my-zsh
drwxr-xr-x   3 USER USER  4096 Sep 27 14:39 /home/USER/.oh-my-zsh/cache
drwxr-xr-x   2 USER USER  4096 Sep 27 14:39 /home/USER/.oh-my-zsh/cache/completions
drwxr-xr-x 303 USER USER 12288 Sep 27 14:39 /home/USER/.oh-my-zsh/plugins
drwxr-xr-x   2 USER USER  4096 Sep 27 14:39 /home/USER/.oh-my-zsh/plugins/git

[oh-my-zsh] For safety, we will not load completions from these directories until
[oh-my-zsh] you fix their permissions and ownership and restart zsh.
[oh-my-zsh] See the above list for directories with group or other writability.

[oh-my-zsh] To fix your permissions you can do so by disabling
[oh-my-zsh] the write permission of "group" and "others" and making sure that the
[oh-my-zsh] owner of these directories is either root or your current user.
[oh-my-zsh] The following command may help:
[oh-my-zsh]     compaudit | xargs chmod g-w,o-w

[oh-my-zsh] If the above didn't help or you want to skip the verification of
[oh-my-zsh] insecure directories you can set the variable ZSH_DISABLE_COMPFIX to
[oh-my-zsh] "true" before oh-my-zsh is sourced in your zshrc file.

******************************************************************************************

I ran the code  compaudit | xargs chmod g-w,o-w and restarted my computer.  Now I get this error message when starting Ubuntu in terminal:
*******************************************************************************************
/home/USER/.zshrc:source:76: permission denied: /home/USER/.oh-my-zsh/oh-my-zsh.sh
********************************************************************************************

Fortunately, if I enter sudo -s, after I enter my password, I can still get to my home directory, but I want to fix it so I can open Ubuntu directly into my home as I previously could.  Please help.

**Comment by mcornella** [comment, marked as answer]
Your `.oh-my-zsh` folder has broken permissions, you need to `chmod 755 ~/.oh-my-zsh` so that zsh works correctly. 

Next, when running Oh My Zsh as root, I suggest you set `ZSH_DISABLE_COMPFIX=true` in your zshrc file (some line before OMZ is sourced) so that you don't get the above error. The error appears because those folders are writable by users other than root. That makes sense since the folders are owned by your user, so in this case you can ignore the error.

You can set the lines below in your zshrc file, before OMZ is sourced:

```zsh
# put this line before the `source $ZSH/oh-my-zsh.sh line`
[[ $USER = root ]] && ZSH_DISABLE_COMPFIX=true
```
